src/lib/modules/aeris-metadata-blocks/submodules/aeris-metadata-single-file-download/components/aeris-metadata-single-file-download.vue

```diff
@@ -17,257 +17,193 @@
       default: "en"
-    }
-  },
-
-  mounted: function() {
-    var event = new CustomEvent("aerisThemeRequest", {});
-    document.dispatchEvent(event);
-  },
-
-  watch: {
-    lang(value) {
-      this.$i18n.locale = value;
     },
-
-    isInCart(value) {
-      this.ensureTheme();
+    theme: {
+      type: Object,
+      default: null
+    },
+    metadata: {
+      type: Object,
+      default: null
+    },
+    isInCart: {
+      type: Boolean,
+      default: false
     }
   },
 
-  destroyed: function() {
-    document.removeEventListener(
-      "aerisMetadataRefreshed",
-      this.aerisMetadataListener
-    );
-    this.aerisMetadataListener = null;
-
-    document.removeEventListener(
-      "cartContentResponse",
-      this.cartContentResponseListener
-    );
-    this.cartContentResponseListener = null;
-
-    document.removeEventListener("aerisTheme", this.aerisThemeListener);
-    this.aerisThemeListener = null;
-
-    document.removeEventListener("recordToAdd", this.recordToAddListener);
-    this.recordToAddListener = null;
-  },
-
-  created: function() {
-    console.log("Aeris single file download - Creating");
-    this.$i18n.locale = this.lang;
-    // to get the datas
-    this.aerisMetadataListener = this.handleRefresh.bind(this);
-    document.addEventListener(
-      "aerisMetadataRefreshed",
-      this.aerisMetadataListener
-    );
-    // to apply theme
-    this.aerisThemeListener = this.handleTheme.bind(this);
-    document.addEventListener("aerisTheme", this.aerisThemeListener);
-    // to get the cart response, asking if the collection is already in the cart
-    this.cartContentResponseListener = this.cartContentResponse.bind(this);
-    document.addEventListener(
-      "cartContentResponse",
-      this.cartContentResponseListener
-    );
-    // as the event is received by aeris-catalog-ui-button, we have to use a custom event to get it back
-    this.recordToAddListener = this.addToCart.bind(this);
-    document.addEventListener("recordToAdd", this.recordToAddListener);
-  },
-
-  updated: function() {
-    this.ensureTheme();
-  },
-
-  computed: {},
-
   data() {
     return {
-      visible: false,
-      theme: null,
-      aerisThemeListener: null,
-      aerisMetadataListener: null,
-      cartResponseListener: null,
-      data: {},
       downloadEntry: {},
       service: null,
       id: null,
-      collectionName: null,
-      isInCart: false,
-      classesButton: "metadata-text-btn"
+      collectionName: null
     };
   },
 
-  methods: {
-    cartContentResponse: function(e) {
-      this.isInCart = false;
-      var cartContent = e.detail.cartContent;
-      if (cartContent) {
-        for (var i = 0; i < cartContent.length; i++) {
-          var cartItem = cartContent[i];
-          if (cartItem.collectionId == this.id) {
-            this.isInCart = true;
-          }
-        }
-      }
+  computed: {
+    isVisible() {
+      return this.service !== null;
     },
+    applyTheme() {
```

Comment:
On peut économiser quelques lignes avec un opérateur ternaire :
this.theme && this.theme.primaryColor ? "--primaryColor": this.theme.primaryColor : "";

Reply:
oui :-P